src/langrila/memory/s3.py

```diff
@@ -0,0 +1,84 @@
+import os
+import logging
+import json
+from typing import Optional, Union
+
+import boto3
+from botocore.exceptions import ClientError
+from botocore.client import Config as BotoConfig
+
+from ..base import BaseConversationMemory
+
+class S3ConversationMemory(BaseConversationMemory):
+    def __init__(
+            self,
+            bucket: str,
+            key: str,
+            *,
```

Comment:
```suggestion
            *,
            key: Optional[str] = None,
```

It might be better to initialize key by first conversation if key isn't given, and also be better to change argument name to `object_key`. What do you think? (See other comments)

Reply:
I thought users should want to manage object keys by own (like me), but looks much better to be an option. will fix.